### PR TITLE
Add shaders

### PR DESCRIPTION
**Pull request by bjoeris**
This PR populates the "Shaders" tab for the currently selected event.

image

### CODE REVIEW

File: dive_core/shader_disassembly.cpp [hunk 1]
```diff
@@ -30,16 +37,52 @@ bool Disassemble(const uint8_t*                             shader_memory,
...
 {
-    return "";
+    std::string         disasm;
+    struct shader_stats stats;
+    {
+        enum debug_t debug = PRINT_RAW;
+        disasm_a3xx_set_debug(debug);
+
+        // TODO(bjoeris): this should come from the file instead of hard-coded.
+        uint32_t chip = 0x43050b00;
```
Comment (wangra-google) [inline dive_core/shader_disassembly.cpp:49]: i added g_sGPU_variant in pm4_info.cpp (generated from generatePm4Info_adreno.py) we could probably add chip_id there as well

**Review by wangra-google — APPROVED** [review]

**Comment by wangra-google** [comment]
> This PR populates the "Shaders" tab for the currently selected event.
> 
> image

Great job! i cannot see from the screen shot, but is the stats from `shaderdb` also shown in the UI?

**Comment by bjoeris** [comment]
> is the stats from `shaderdb` also shown in the UI?

Do you mean the stats shown after the disassembly in cffdump? E.g.
```
Stats:
- shaderdb: 56 instr, 9 nops, 47 non-nops, 2 mov, 0 cov
- shaderdb: 0 last-baryf, 0 half, 3 full, 4 constlen
- shaderdb: 17 cat0, 3 cat1, 8 cat2, 24 cat3, 0 cat4, 0 cat5, 4 cat6, 0 cat7
- shaderdb: 0 sstall, 4 (ss), 2 (sy)
```

I'm not currenly displaying these, except for "full", which as far as I can tell is the number of GPRs used by the shader.

**Comment by wangra-google** [comment]
> > is the stats from `shaderdb` also shown in the UI?
> 
> Do you mean the stats shown after the disassembly in cffdump? E.g.
> 
> ```
> Stats:
> - shaderdb: 56 instr, 9 nops, 47 non-nops, 2 mov, 0 cov
> - shaderdb: 0 last-baryf, 0 half, 3 full, 4 constlen
> - shaderdb: 17 cat0, 3 cat1, 8 cat2, 24 cat3, 0 cat4, 0 cat5, 4 cat6, 0 cat7
> - shaderdb: 0 sstall, 4 (ss), 2 (sy)
> ```
> 
> I'm not currenly displaying these, except for "full", which as far as I can tell is the number of GPRs used by the shader.

is it difficult to add those? i think it would be interesting to show those states (not necessarily in this PR though)

**Comment by wangra-google** [comment]
> > is it difficult to add those? i think it would be interesting to show those states (not necessarily in this PR though)
> 
> It's not difficult to add, but where should they go in the UI? One option is to just add them as text after the disassembly (similar to the output from cffdump).

yeah, i think that should be good enough

**Review by wangra-google — DISMISSED** [review]

**Review by wangra-google — APPROVED** [review]

**Pull request merged**